crates/starknet-devnet/tests/test_messaging.rs

```diff
@@ -660,4 +669,117 @@ mod test_messaging {
+            MESSAGING_L1_ADDRESS
+        );
+
+        // Deploy the L1L2 testing contract on L1 (on L2 it's already pre-deployed).
+        let l1_messaging_address = H160::from_str(MESSAGING_L1_ADDRESS).unwrap();
+        let eth_l1l2_address = anvil.deploy_l1l2_contract(l1_messaging_address).await.unwrap();
+        let eth_l1l2_address_hex = format!("{eth_l1l2_address:#x}");
+
+        let eth_l1l2_address_felt = felt_from_prefixed_hex(&eth_l1l2_address_hex).unwrap();
+        let user_sn = Felt::ONE;
+        let user_eth: U256 = 1.into();
+
+        // Set balance to for the user 1 on L2.
+        let user_balance = Felt::TWO;
+        increase_balance(sn_account.clone(), sn_l1l2_contract, user_sn, user_balance).await;
+        assert_eq!(get_balance(&devnet, sn_l1l2_contract, user_sn).await, [user_balance]);
+
+        // Withdraw on L2 to send it to L1 with a l2->l1 message.
+        withdraw(
+            sn_account.clone(),
+            sn_l1l2_contract,
+            user_sn,
+            user_balance,
+            eth_l1l2_address_felt,
+        )
+        .await;
+        assert_eq!(get_balance(&devnet, sn_l1l2_contract, user_sn).await, [Felt::ZERO]);
+
+        // Flush to send the messages.
+        devnet.send_custom_rpc("devnet_postmanFlush", json!({})).await.unwrap();
+
+        // Check that the balance is 0 on L1 before consuming the message.
+        let user_balance_eth = anvil.get_balance_l1l2(eth_l1l2_address, user_eth).await.unwrap();
+        assert_eq!(user_balance_eth, 0.into());
+
+        let sn_l1l2_contract_u256 = felt_to_u256(sn_l1l2_contract);
+
+        // Consume the message to increase the balance.
+        anvil
+            .withdraw_l1l2(
+                eth_l1l2_address,
+                sn_l1l2_contract_u256,
+                user_eth,
+                felt_to_u256(user_balance),
+            )
+            .await
+            .unwrap();
+
+        let user_balance_eth = anvil.get_balance_l1l2(eth_l1l2_address, user_eth).await.unwrap();
+        assert_eq!(user_balance_eth, felt_to_u256(user_balance));
+
+        // Send back the amount 1 to the user 1 on L2.
+        let deposit_amount = 1;
+        anvil
+            .deposit_l1l2(eth_l1l2_address, sn_l1l2_contract_u256, user_eth, deposit_amount.into())
+            .await
+            .unwrap();
+
+        let user_balance_eth = anvil.get_balance_l1l2(eth_l1l2_address, user_eth).await.unwrap();
+
+        assert_eq!(user_balance_eth, 1.into()); // 2 - 1
+        assert_eq!(get_balance(&devnet, sn_l1l2_contract, user_sn).await, [Felt::ZERO]);
+
+        // Flush messages to have MessageToL2 executed.
+        devnet.send_custom_rpc("devnet_postmanFlush", json!({})).await.unwrap();
+
+        assert_eq!(user_balance_eth, deposit_amount.into());
+        assert_eq!(get_balance(&devnet, sn_l1l2_contract, user_sn).await, [Felt::ONE]);
+
+        devnet.send_custom_rpc("devnet_postmanFlush", json!({})).await.unwrap();
```

Comment:
why this second flush is needed?

Reply:
I was testing redundant flushing while developing to see the behavior, but I guess it's not needed anymore, so I removed the line: https://github.com/0xSpaceShard/starknet-devnet-rs/pull/609/commits/911e8b9bd056a109e0a189183743f74370b8ca3f